src/internal/group_api/mod.rs

```diff
@@ -297,47 +359,81 @@ pub(crate) fn get_group_keys<'a>(
+                use std::{collections::HashSet, iter::FromIterator};
+                let desired_members_set: HashSet<&UserId> = HashSet::from_iter(members);
+                let found_members: Vec<UserId> =
+                    user_ids_and_keys.into_iter().map(|(x, _)| x).collect();
+                let found_members_set: HashSet<&UserId> = HashSet::from_iter(&found_members);
+                let diff: HashSet<&&UserId> =
+                    desired_members_set.difference(&found_members_set).collect();
+                futures::future::err(IronOxideErr::UserDoesNotExist(format!(
+                    "Failed to find the following users from the `members` list: {:?}",
+                    diff
+                )))
+            } else {
+                transform::gen_group_keys(recrypt)
+                    .and_then(move |(plaintext, group_priv_key, group_pub_key)| {
+                        // encrypt the group secret to the current user as they will be the group admin
+                        let encrypted_group_key = recrypt.encrypt(
+                            &plaintext,
+                            &user_master_pub_key.into(),
+                            &auth.signing_private_key().into(),
+                        )?;
+                        // Map from UserId to (PublicKey, Optional TransformKey)
+                        // TransformKey is optional because we only need it for members, but won't need it
+                        // when we expand this to include admins
+                        let maybe_user_info: Result<
+                            HashMap<UserId, (PublicKey, Option<TransformKey>)>,
+                            _,
+                        > = user_ids_and_keys
+                            .into_iter()
+                            .map(|(id, user_pub_key)| {
+                                // this is for when we expand `member_ids_and_keys` to also have admins in it
+                                if members.contains(&id) {
+                                    let maybe_transform_key = recrypt.generate_transform_key(
+                                        &group_priv_key.clone().into(),
+                                        &user_pub_key.clone().into(),
+                                        &auth.signing_private_key().into(),
+                                    );
+                                    maybe_transform_key.map(|transform_key| {
+                                        (id, (user_pub_key, Some(transform_key.into())))
+                                    })
+                                } else {
+                                    Ok((id, (user_pub_key, None)))
+                                }
+                            })
+                            .collect();
+
+                        let user_info = maybe_user_info?;
+
+                        let maybe_user_info = if user_info.len() != 0 {
```

Comment:
Reusing this variable name seems confusing to me. Maybe name the first one `error_or_user_info` or something.

Reply:
I agreed and just changed that